Dataset individual page

luccasmmg opened the pull request.

Gutts-n commented:
@luccasmmg I opened the page without signing in and received this error 

![image](https://github.com/user-attachments/assets/60df3cab-0f28-40a8-80c0-54ddf2ea169d)

Link: https://tdc-data-portal-git-dataset-individual-page-datopian1.vercel.app/Transport%20Data%20Commons/c4a9f489-1d6a-4760-9097-b5866604feb1

Gutts-n commented on frontend/src/server/api/routers/user.ts:
```diff
@@ -143,11 +143,11 @@ export const userRouter = createTRPCRouter({
-      const user = ctx.session.user;
-      const apiKey = user.apikey;
+      const user = ctx.session?.user;
+      const apiKey = user?.apikey ?? env.SYS_ADMIN_API_KEY;
```
Shouldn't you send an empty key when the user is not signed in to return the public data from the CKAN?

luccasmmg replied:
thanks for spotting this, this is no longer needed cause of the backed changes

luccasmmg commented on frontend/src/components/dataset/individualPage/Downloads.tsx:
```diff
@@ -1,116 +1,46 @@
+    api.ga.getDownloadStats.useQuery({
+      id: dataset.id,
+    });
+  const dataResources = dataset.resources.filter(
+    (r) => r.resource_type && r.resource_type === "data"
+  );
+  const docsResources = dataset.resources.filter(
+    (r) => r.resource_type && r.resource_type === "documentation"
+  );
```
Thats very ugly code what you are suggesting

Gutts-n replied:
It's ugly but you won't run through the array twice.

luccasmmg commented:
> @luccasmmg I opened the page without signing in and received this error
> 
> ![image](https://private-user-images.githubusercontent.com/57202549/376368130-60df3cab-0f28-40a8-80c0-54ddf2ea169d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MzcxNjIsIm5iZiI6MTcyODkzNjg2MiwicGF0aCI6Ii81NzIwMjU0OS8zNzYzNjgxMzAtNjBkZjNjYWItMGYyOC00MGE4LTgwYzAtNTRkZGYyZWExNjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDIwMTQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiMGNmNmQyNTVlZjgyMmQyMTAzZTBmMDg3M2E0OWNlNzAxZDU3OTNiNDE3NDQxMWI0NThjYjA3MGNmNjM5ZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5Xipl8xwqoqh-0ldCtBPF1Y5irzBbNVAsGBqvmRxUYs)
> 
> Link: https://tdc-data-portal-git-dataset-individual-page-datopian1.vercel.app/Transport%20Data%20Commons/c4a9f489-1d6a-4760-9097-b5866604feb1

Good catch, fixed, there were two errors

One in the `dataset tab` which should not appear when the dataset doesnt have any datastore_active resources, so i fixed that

The other is in the `metadata` tab, which technically happened because the live deployment doesnt have the updates to `package_show` which i added, nonetheless i added error handling to take care of it, even tough i do not expect the value which was undefined here to ever be undefined once this PR gets merged
